jenkins/ansible/roles/createNightlyBuildJob/files/utils.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+#
+# Authors:
+#     Dinesh Prasanth M K <dmoluguw@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+# All rights reserved.
+#
+
+import requests
+from requests.auth import HTTPBasicAuth
+import datetime
+
+
+class CoprUtil:
+
+    DELETE_RESULT = {
+            204: "was removed successfully",
+            400: "can't be deleted currently",
+            403: "authroization failed",
+            404: "doesn't exists"
+    }
+
+    def __init__(self, client=None):
+        """
+            :param unicode repo: COPR repo to work on
+            :param unicode copr_url: COPR URL to access end points
+        """
+        self.client = client
+        self.copr_api_url = client.api_root or u"https://copr.fedorainfracloud.org/api_2"
+        self.netClient = client.nc
+
+    def sendGetRequest(self, url, params):
+        """
+            :param unicode url: request URL
+            :param json params: params to embed in the GET request
+        """
+        # Send GET request
+        response = requests.get(url = url, params = params)
+
+        # Extract data in JSON format
+        return response.json()
+
+    def getProjectID(self, name=None, group='pki'):
+        """
+            :param unicode searchQuery: search parameter similar to search box in COPR site
+            :param unicode name: repository name to search for
+            :param unicode group: group name of the project
+        """
+        # Construct URL for project endpoint
+        url = "{0}/projects".format(self.copr_api_url)
+
+        # Define params to be sent to the End Point
+        if name and group:
+            PARAMS = {'group': group, 'name': name}
+        else:
+            return None
+
+        receivedData = self.sendGetRequest(url=url, params=PARAMS)
+
+        projects = receivedData['projects']
+        if(len(projects) != 1):
+            raise ValueError("Provide project specific details to get Project ID")
+        else:
+            # Found exact project. Extract project ID
+            project = projects[0]['project']
+            return project['id']
+
+    def findBuildIDs(self, projectID=None, package=None, minAge=None):
+        """
+            :param integer projectID: project ID
+            :param unicode package: package name 
```

Comment:
We want to remove all packages older than certain age, so the package param should not be required.